arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/raise/Builders.kt

```diff
@@ -221,27 +222,30 @@ public class SingletonRaise<in E>(
     map { it.bind() }.toNonEmptySet()
 
   @RaiseDSL
-  public fun <A> recover(
+  public inline fun <A> recover(
     block: SingletonRaise<E>.() -> A,
     recover: () -> A,
   ): A {
     contract {
       callsInPlace(block, InvocationKind.AT_MOST_ONCE)
       callsInPlace(recover, InvocationKind.AT_MOST_ONCE)
     }
-    return recover<E, A>({ block(SingletonRaise(error, this as Raise<Any?>)) }, { _ -> recover() })
+    return singleton(recover) { ignoreErrors(block) }
   }
 
   /**
    * Introduces a scope where you can [bind] errors of any type,
    * but no information is saved in the [raise] case.
    */
   @RaiseDSL
-  public fun <A> ignoreErrors(
-    block: Raise<Any?>.() -> A,
+  public inline fun <A> ignoreErrors(
+    block: SingletonRaise<Any?>.() -> A,
```

Comment:
I'm a bit afraid of this change. If we have `Raise<Any?>`, we can directly use `ignoreErrors` with some callable reference of the right type.

Reply:
Hmmm, I haven't tested this, but wouldn't that just work? Any function that needs `Raise<Any?>` would accept `SingletonRaise<Any?>`. I'll add a test for this and see what happens.

Reply:
Added test! It functions just fine. Refining the type seems to only allow extra functionality.